cmd/kueuectl/app/list/list_workload.go

```diff
@@ -180,7 +191,13 @@ func (o *WorkloadOptions) Run(ctx context.Context) error {
 		return err
 	}
 
-	o.filterList(list)
+	r := newListWorkloadResources()
```

Comment:
This is doing a list of **all** Workloads before filtering.

Could we instead do a single get of the owner object, grab the UID and use the label `kueue.x-k8s.io/job-uid` in a label selector?

Reply:
It's problematic for Composable jobs.

Reply:
We could specialize it for composable jobs. In general, List calls are slow and can lead to apiserver starvation.

Reply:
Done. Using unstructured objects allowed to deal with different job types.

Reply:
> We could specialize it for composable jobs. In general, List calls are slow and can lead to apiserver starvation.

The specialization will need to be on some kind of hardcoded list.. or we can always do label filtered list and if that list is empty do a second full list with client side filtering.

Reply:
@alculquicondor WDYT?

Reply:
Yes, doing a full list if all other attempts failed seems ok.

Reply:
as for getting the object UID, I hope we can do it in a generic way (it should be possible using the dynamic client https://caiorcferreira.github.io/post/the-kubernetes-dynamic-client/).

Reply:
> as for getting the object UID, I hope we can do it in a generic way (it should be possible using the dynamic client https://caiorcferreira.github.io/post/the-kubernetes-dynamic-client/).

This is done. The part left is filtering composable jobs.

Reply:
Composable jobs filtering is done too.